src/core/Table/Table.test.tsx

```diff
@@ -2227,3 +2227,193 @@ it('should sync body horizontal scroll with header scroll', () => {
+        data={mockedData()}
+        emptyTableContent='Empty table'
+        emptyFilteredTableContent='No results. Clear filter.'
+        enableDraggableColumns
+        isSortable
+        onSort={onSort}
+      />,
+    );
+
+    const headerCells = container.querySelectorAll<HTMLDivElement>(
+      '.iui-table-header .iui-cell',
+    );
+    headerCells.forEach((cell) =>
+      expect(cell.getAttribute('draggable')).toBe('true'),
+    );
+
+    const srcColumn = headerCells[srcIndex];
+    const dstColumn = headerCells[dstIndex];
+
+    // Helper function to create a drag event because regular ones does not pass `clientX`
+    const createBubbledEvent = (type: string, props = {}) => {
+      const event = new Event(type, { bubbles: true });
+      Object.assign(event, props);
+      return event;
+    };
+
+    const setData = jest.fn();
+    const clientX = dropPosition === 'right' ? 75 : 25;
+
+    srcColumn.dispatchEvent(
+      createBubbledEvent('dragstart', { dataTransfer: { setData } }),
+    );
+    expect(setData).toHaveBeenCalledWith('text', srcName);
+    dstColumn.dispatchEvent(createBubbledEvent('dragenter', { clientX }));
+    dstColumn.dispatchEvent(createBubbledEvent('dragover', { clientX }));
+    expect(dstColumn).toHaveClass('iui-reorder-column-' + dropPosition);
+    dstColumn.dispatchEvent(
+      createBubbledEvent('drop', {
+        clientX,
+        dataTransfer: { getData: () => srcName },
+      }),
+    );
+
+    // Should not trigger sort
+    expect(onSort).not.toHaveBeenCalled();
+
+    rerender(
+      <Table
+        columns={columns()}
+        data={mockedData()}
+        emptyTableContent='Empty table'
+        emptyFilteredTableContent='No results. Clear filter.'
+        enableDraggableColumns
+        isSortable
+        onSort={onSort}
+      />,
+    );
+
+    container
+      .querySelectorAll<HTMLDivElement>('.iui-table-header .iui-cell')
+      .forEach((cell, index) =>
+        expect(cell.textContent).toBe(resultingColumns[index]),
+      );
+  },
+);
+
+it('should not have `draggable` attribute on columns with `disableDragging` enabled', () => {
+  const columns: Column<TestDataType>[] = [
+    {
+      Header: 'Header name',
+      columns: [
+        {
+          id: 'name',
+          Header: 'Name',
+          accessor: 'name',
+        },
+        {
+          id: 'description',
+          Header: 'description',
+          accessor: 'description',
+        },
+        {
+          id: 'view',
+          Header: 'view',
+          Cell: () => 'View',
+          disableDragging: true,
+        },
+      ],
+    },
+  ];
+  const { container } = render(
+    <Table
+      columns={columns}
+      data={mockedData()}
+      emptyTableContent='Empty table'
+      emptyFilteredTableContent='No results. Clear filter.'
+      enableDraggableColumns
+      isSelectable
+      subComponent={(row) => (
+        <div>{`Expanded component, name: ${row.original.name}`}</div>
+      )}
+    />,
+  );
+
```

Comment:
Tried to mock drag n drop events and in jsdom they just work, so I guess wit is bug in jsdom because in real browser you can't drag if `draggable` isn't present.